lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -3661,7 +3663,7 @@ struct DotOpConversion : public ConvertTritonGPUOpToLLVMPattern<triton::DotOp> {
 
     // Here we assume the DotOp's operands always comes from shared memory.
     auto AShape = A.getType().cast<RankedTensorType>().getShape();
-    size_t reduceAxis = 1;
+    size_t reduceAxis = op.transA() ? 0 : 1;
```

Comment:
This is not covered in test_gemm.py. Will it PASS if we simpliy add some tests in test_gemm.py with trans_a/b in tl.dot? If it fails it's still better to at least add some TODO.

Reply:
Here is a missing point when porting the code.
I will add some test in the following PR, still can not run with some issue already know, for example, the transA/transB cannot be accessed in convertLayout[shared->dot_op], this is another storey.